src/as_list.cpp

```diff
@@ -0,0 +1,34 @@
+#include "r_trie.h"
+
+template <typename T>
+static inline Rcpp::List trie_to_list(SEXP radix){
+  r_trie <T>* rt_ptr = (r_trie <T> *) R_ExternalPtrAddr(radix);
+  ptr_check(rt_ptr);
+  
+  return Rcpp::List::create(
+    Rcpp::Named("keys") = rt_ptr->get_keys(),
```

Comment:
Couldn't we just do this super-trivially R-side, without even needing methods, through:

> as.list.trie <- function(trie){
   return(list(keys = get_keys(trie),
                   values = get_values(trie)))
}

? Or I'm missing something

(Similarly we could add as.data.frame and as.vector, with the second returning a vector of the values with the keys as names. I should open an issue for those)